packages/@monocdk-experiment/rewrite-imports/bin/rewrite-imports.ts

```diff
@@ -1,237 +1,36 @@
-
-  const cdkImports = new Array<ts.Statement>();
-  if (namespaces.length > 0) {
-    cdkImports.push(ts.createImportDeclaration(
-      /* decorators */ undefined,
-      /* modifiers */ undefined,
-      ts.createImportClause(
-        /* name */ undefined,
-        ts.createNamedImports(
-          namespaces.map(nsImport => ts.createImportSpecifier(ts.createIdentifier(nsImport.ns), ts.createIdentifier(nsImport.alias ?? nsImport.ns))),
-        )
-      ),
-      ts.createStringLiteral('monocdk-experiment')));
-  }
-
-  const cdkRebinds = new Array<ts.Statement>();
-  for (const [nsName, symbols] of Object.entries(symbolsByNamespace)) {
-    cdkRebinds.push(ts.createVariableStatement(
-      [], [
-        ts.createVariableDeclaration(
-          ts.createObjectBindingPattern(symbols.map(sym => ts.createBindingElement(
-            /* dotDotDotToken */ undefined,
-            undefined,
-            ts.createIdentifier(sym.symbolName),
-            ))),
-          /* type */ undefined,
-          ts.createIdentifier(nsName)
-        )
-      ]));
-  }
-
-  const newStatements = [
-    ...cdkImports,
-    ...otherImports,
-    ...cdkRebinds,
+  const ignore = [
+    '**/*.d.ts',
+    'node_modules/**'
   ];
 
-  return [newStatements, cdkImports.length + cdkRebinds.length > 0];
-}
-
-function analyzeCdkImport(node: ImportNode): ImportAnalysis | undefined {
-  if (ts.isImportDeclaration(node)) {
-    if (ts.isStringLiteral(node.moduleSpecifier)) {
-      const cdkSubModule = extractCdkModuleName(node.moduleSpecifier.text);
-      if (cdkSubModule === undefined) { return undefined; }
-
-      if (node.importClause) {
-        if (node.importClause.name) {
-          return { cdkSubModule, type: 'module', moduleAlias: node.importClause.name.text };
-        }
-        if (node.importClause.namedBindings) {
-          if (ts.isNamespaceImport(node.importClause.namedBindings)) {
-            return { cdkSubModule, type: 'module', moduleAlias: node.importClause.namedBindings.name.text };
-          }
-          if (ts.isNamedImports(node.importClause.namedBindings)) {
-            return {
-              cdkSubModule,
-              type: 'symbols',
-              symbols: node.importClause.namedBindings.elements.map(el => ({
-                name: el.name.text,
-                alias: el.propertyName?.text,
-              }))
-            };
-          }
-        }
-      }
-    }
-  }
-  if (ts.isImportEqualsDeclaration(node)) {
-    if (ts.isExternalModuleReference(node.moduleReference)) {
-      if (ts.isStringLiteral(node.moduleReference.expression)) {
-        const cdkSubModule = extractCdkModuleName(node.moduleReference.expression.text);
-        if (cdkSubModule === undefined) { return undefined; }
-
-        return { cdkSubModule, type: 'module', moduleAlias: node.name.text };
-      }
+  const files = await glob(process.argv[2], { ignore, matchBase: true });
+  for (const file of files) {
+    const input = await readFile(file, 'utf-8');
```

Comment:
It would be nice to avoid `await` in `for` loops as this actually misses on the opportunity to parallelize IO. In order to do this, you would map `files` to `Promise<T>` and then use `await Promise.all` of these.

This is a micro-optimization (while I would like to generally speaking get rid of all `await-in-loop`, I'm not sure the gains are going to be so huge), so feel free to leave it!

Reply:
I am leaving it if that's okay with you for the benefit of readability